src/text/span.rs

```diff
@@ -562,6 +565,7 @@ mod tests {
         assert_eq!(Span::raw("").width(), 0);
         assert_eq!(Span::raw("test").width(), 4);
         assert_eq!(Span::raw("test content").width(), 12);
+        assert_eq!(Span::raw("test\ncontent").width(), 12);
```

Comment:
Regarding this - we likely need to filter control characters in the width method to avoid this. It would be ideal if the width method returns the display width and not the unicode width. The crate recently changed to make newlines take up a single character per the unicode spec, but that's not really useful for rendering to the terminal.

Reply:
Agree, filed https://github.com/ratatui-org/ratatui/issues/1271 to unblock this PR.

Reply:
It's worth putting a comment on this line to note it as possibly wrong. and link to #1271